TAS.Avalonia/Communication/StudioCommunicationServer.cs

```diff
@@ -146,7 +146,24 @@ public class StudioCommunicationServer : StudioCommunicationBase {
     public void ConvertToLibTas(string path) => PendingWrite = () => ConvertToLibTasNow(path);
     public void SendHotkeyPressed(HotkeyID hotkey, bool released = false) => PendingWrite = () => SendHotkeyPressedNow(hotkey, released);
     public void ToggleGameSetting(string settingName, object value) => PendingWrite = () => ToggleGameSettingNow(settingName, value);
-    public void GetDataFromGame(GameDataType gameDataType, object arg) => PendingWrite = () => GetGameDataNow(gameDataType, arg);
+    public void RequestDataFromGame(GameDataType gameDataType, object arg) => PendingWrite = () => RequestGameDataNow(gameDataType, arg);
+
+    public string GetDataFromGame(GameDataType gameDataType, object arg = null) {
+        CommunicationWrapper.ReturnData = null;
+        RequestDataFromGame(gameDataType, arg);
+
+        int sleepTimeout = 150;
+        while (CommunicationWrapper.ReturnData == null && sleepTimeout > 0) {
+            Thread.Sleep(10);
```

Comment:
Not a big fan of `Thread.Sleep` but it'll do for now.

Reply:
Simply copied that from Studio which does Thread.Sleep as well.

Reply:
Well Studio is wrong. ;)
It's fine for now, we'll look at proper async and `Task.Delay` later.